pkg/git/git.go

```diff
@@ -46,6 +46,25 @@ type Options struct {
 	Email  string
 }
 
+// ValidateURL validates the URL field of this Options object, returning an
+// error should the current value not be valid.
+func (o Options) ValidateURL() error {
+	if o.URL == "" {
+		return errors.New("empty Git URL")
+	}
+	if !IsGitURL(o.URL) {
+		return errors.New("invalid Git URL")
+	}
+	if !o.isSSHURL() {
+		return errors.New("got a HTTP(S) Git URL, but eksctl currently only supports SSH Git URLs")
+	}
+	return nil
+}
+
+func (o Options) isSSHURL() bool {
+	return strings.HasPrefix(o.URL, "git@") || strings.HasPrefix(o.URL, "ssh://")
```

Comment:
For more accurate parsing, we should use the same library call `IsGitURL` uses: `giturls.Parse`
and check for the `ssh` scheme.

Reply:
`IsGitURL` is already called in a previous check, before reaching this part.

Reply:
I did see that. I was suggesting to somehow use the URL from that call and check its scheme.

Reply:
I'd rather keep the two functions separate for now, but I'm doing this now: https://github.com/weaveworks/eksctl/pull/1232/commits/efbf581aa04517d010a73d3278f419288c141801 to avoid reinventing the wheel and potentially missing edge cases.

Reply:
I'd rather keep the two functions separate for now, but I'm doing this now: https://github.com/weaveworks/eksctl/pull/1232/commits/efbf581aa04517d010a73d3278f419288c141801 to avoid reinventing the wheel and potentially missing edge cases. WDYT?

Reply:
FYI: the SSH URL used in the tests was having a scheme set to `"git"`, hence the current conditionals.